### PR TITLE
Use robust install script from get.secrethub.io

### PR DESCRIPTION
**Pull request by florisvdg**
Fixing possible collisions because it doesn't untar the CLI in the working directory anymore and also increasing OS and shell compatibility.

### CODE REVIEW

File: src/commands/install.yml [hunk 1]
```diff
@@ -18,39 +18,62 @@ steps:
...
         else
-            PLATFORM=linux
+            echo -e "${ERROR_COLOR}Cannot determine OS type. Exiting...${NO_COLOR}"
+            exit;
+        fi
+
+        # Make sure we have root priviliges.
+        SUDO=""
+        if [ $(id -u) -ne 0 ]; then
+            if ! [ $(command -v sudo) ]; then
```
Comment (SimonBarendse) [inline src/commands/install.yml:53]: Should we remove this check as well now? You've already removed the `sudo -v` command below.

File: src/commands/install.yml [hunk 2]
```diff
@@ -18,39 +18,62 @@ steps:
...
+        $SUDO mkdir -p /usr/local/secrethub/bin
+
+        # Retrieve latest version
+        if [ -n "${SECRETHUB_CLI_VERSION}" ]; then
+          VERSION=v${SECRETHUB_CLI_VERSION}
+        else
+          echo -e "${OK_COLOR}==> Retrieving latest version${NO_COLOR}"
+          VERSION=$(curl --silent "https://api.github.com/repos/secrethub/secrethub-cli/releases/latest" | grep tag_name | awk -F\" '{ print $4 }')
         fi
+
```
Comment (SimonBarendse) [inline src/commands/install.yml:71]: This previous iteration of the install script had a check to not re-install if the requested version is already installed. Do we want to keep that?

**Review by SimonBarendse — COMMENTED** [review]
Let's change the PR title to reflect what this PR does for this repo, instead of just reflecting how it was implemented.

File: src/commands/install.yml (hunk 1 above)
Comment (florisvdg) [reply src/commands/install.yml:53]: This still has a purpose: checking if you even have `sudo`.

File: src/commands/install.yml (hunk 2 above)
Comment (florisvdg) [reply src/commands/install.yml:71]: For now I'd just prefer a more consistent and robust install script. Let's add that later on.

File: src/commands/install.yml (hunk 1 above)
Comment (SimonBarendse) [reply src/commands/install.yml:53]: Ah I get it know. If you're already root, we don't use `sudo`, as the `SUDO` variable is set to the empty string. So in that case, you also don't need the `sudo` command. Thanks

File: src/commands/install.yml (hunk 2 above)
Comment (SimonBarendse) [reply src/commands/install.yml:71]: I don't see the need to wait. We already have the code:

```sh
if command -v secrethub >/dev/null 2>&1 && [[ $(secrethub --version 2>&1 | cut -d "," -f 1) = *"$VERSION" ]]; then
  exit 0
fi
```

**Review by SimonBarendse — APPROVED** [review]

**Pull request merged**